src/xb-node.c

```diff
@@ -258,6 +282,134 @@ xb_node_get_children (XbNode *self)
 	return array;
 }
 
+/**
+ * xb_node_child_iter_init:
+ * @iter: an uninitialized #XbNodeChildIter
+ * @self: a #XbNode
+ *
+ * Initializes a child iterator for the node's children and associates
+ * it with @self.
+ * The #XbNodeChildIter structure is typically allocated on the stack
+ * and does not need to be freed explicitly.
+ *
+ * Since: 0.3.4
+ */
+void
+xb_node_child_iter_init (XbNodeChildIter *iter, XbNode *self)
+{
+	XbNodePrivate *priv = GET_PRIVATE (self);
+	RealChildIter *ri = (RealChildIter *) iter;
+
+	g_return_if_fail (iter != NULL);
+	g_return_if_fail (XB_IS_NODE (self));
+
+	ri->node = self;
+	ri->position = xb_silo_get_child_node (priv->silo, priv->sn);
+	ri->first_iter = TRUE;
+}
+
+/**
+ * xb_node_child_iter_next:
+ * @iter: an initialized #XbNodeAttrIter
+ * @child: (out) (optional) (not nullable): Destination of the returned child
+ *
+ * Returns the current child and advances the iterator.
+ * The retrieved #XbNode child needs to be dereferenced with g_object_unref().
+ * Example:
+ * |[<!-- language="C" -->
+ * XbNodeChildIter iter;
+ * g_autoptr(XbNode) child = NULL;
+ *
+ * xb_node_child_iter_init (&iter, node);
+ * while (xb_node_child_iter_next (&iter, &child)) {
+ *     // do something with the node child
+ *     g_clear_pointer (&child, g_object_unref);
+ * }
+ * ]|
+ *
+ * Returns: %FALSE if the last child has been reached.
+ *
+ * Since: 0.3.4
+ */
+gboolean
+xb_node_child_iter_next (XbNodeChildIter *iter, XbNode **child)
+{
+	XbNodePrivate *priv;
+	RealChildIter *ri = (RealChildIter *) iter;
+
+	g_return_val_if_fail (iter != NULL, FALSE);
+	g_return_val_if_fail (child != NULL, FALSE);
+	priv = GET_PRIVATE (ri->node);
+
+	/* check if the iteration was finished */
+	if (ri->position == NULL) {
+		*child = NULL;
+		return FALSE;
+	}
+
+	*child = xb_silo_create_node (priv->silo, ri->position, FALSE);
+	ri->position = xb_silo_get_next_node (priv->silo, ri->position);
+
+	return TRUE;
+}
+
+/**
+ * xb_node_child_iter_loop: (skip)
+ * @iter: an initialized #XbNodeAttrIter
+ * @child: (out) (optional) (not nullable): Destination of the returned child
```

Comment:
```suggestion
 * @child: (out) (optional) (nullable): Destination of the returned child
```

It is nullable on the last iteration